src/WebJobs.Script.WebHost/Middleware/CustomHeadersMiddleware.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.WebJobs.Script.Middleware;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Middleware
+{
+    public class CustomHeadersMiddleware : IJobHostHttpMiddleware
+    {
+        private RequestDelegate _invoke;
+
+        public CustomHeadersMiddleware(IOptions<ScriptJobHostOptions> hostOptions)
+        {
+            RequestDelegate contextNext = async context =>
+            {
+                if (context.Items.Remove(ScriptConstants.CustomHeadersMiddlewareRequestDelegate, out object requestDelegate) && requestDelegate is RequestDelegate next)
+                {
+                    await next(context);
+
+                    foreach (var header in hostOptions.Value.CustomHeaders)
+                    {
+                        context.Response.Headers[header.Key] = header.Value;
```

Comment:
Wondering if we should use `TryAdd` here... that would give the function the flexibility to override the header in configuration, if needed.